pkg/Cpanel/Security/Advisor/Assessors/Trojans.pm

```diff
@@ -73,7 +91,618 @@ sub _check_for_libkeyutils {
+            $signature   = $check_bin . ": \"" . $1 . "\"";
+            $has_cdorked = 1;
+            last;
+        }
+    }
+    if ( $has_cdorked == 1 ) {
+        $self->add_bad_advice(
+            'text'       => [ "CDORKED rootkit check: Evidence of the CDORKED A rootkit was found." ],
+            'suggestion' => [
+                'Check the following for more information "[output,url,_1,We Live Security More Information,_2,_3]"',
+                'http://www.welivesecurity.com/wp-content/uploads/2014/03/operation_windigo.pdf',
+                'target',
+                '_blank'
+            ],
+        );
+    }
+}
+
+sub _check_for_cdorked_B_rootkit {
+    my ($self) = @_;
+    my $has_cdorked_b = 0;
+    my @files = ( '/usr/sbin/arpd ', '/usr/sbin/tunelp ', '/usr/bin/s2p ' );
+    my $cdorked_files;
+
+    for my $file (@files) {
+        if ( -e $file ) {
+            $has_cdorked_b = 1;
+            $cdorked_files .= "[$file] ";
+        }
+    }
+
+    if ( $has_cdorked_b == 1 ) {
+        $self->add_bad_advice(
+            'text'       => [ "CDORKED rootkit check: Evidence of the CDORKED B rootkit was found." ],
+            'suggestion' => [
+                'Check the following for more information "[output,url,_1,We Live Security More Information,_2,_3]"',
+                'http://www.welivesecurity.com/wp-content/uploads/2014/03/operation_windigo.pdf',
+                'target',
+                '_blank'
+            ],
+        );
+    }
+}
+
+sub _check_for_libkeyutils_filenames {
+    my ($self) = @_;
+    my $bad_libs;
+    my @dirs  = qw( /lib /lib64 );
+    my @files = qw(
+      libkeyutils.so.1.9
+      libkeyutils-1.2.so.0
+      libkeyutils-1.2.so.2
+      libkeyutils.so.1.3.0
+      libkeyutils.so.1.3.2
+      libns2.so
+      libns5.so
+      libpw3.so
+      tls/libkeyutils.so.1
+      tls/libkeyutils.so.1.5
+    );
+
+    for my $dir (@dirs) {
+        next if !-e $dir;
+        for my $file (@files) {
+            if ( -f "${dir}/${file}" and !-z "${dir}/${file}" ) {
+                $bad_libs .= "\t${dir}/${file}\n";
+            }
+        }
+    }
+
+    if ($bad_libs) {
+        $self->add_bad_advice(
+            'text'       => [ "Libkey rootkit check: suspicious system libraries were found that could indicate a root level compromise." ],
+            'suggestion' => [
+                'Check the following for more information "[output,url,_1,We Live Security More Information,_2,_3]"',
+                'http://www.welivesecurity.com/wp-content/uploads/2014/03/operation_windigo.pdf',
+                'target',
+                '_blank'
+            ],
+        );
+    }
+}
+
+sub build_libkeyutils_file_list {
+    my ($self) = @_;
+    my @dirs = qw( /lib /lib64 );
+    my @libkeyutils_files;
+
+    for my $dir (@dirs) {
+        next if !-e $dir;
+        opendir( my $dir_fh, $dir );
+        while ( my $file = readdir($dir_fh) ) {
+            if ( $file =~ /^libkeyutils\.so\.(?:[\.\d]+)?$/ ) {
```

Comment:
With this regex, the file must either end in a period or with a number.  Did you mean to do the following instead?

```perl
$file =~ /^libkeyutils\.so(?:\.[\.\d]+)?$/
```